src/uct/tcp/tcp_iface.c

```diff
@@ -218,6 +224,55 @@ static int uct_tcp_iface_is_reachable(const uct_iface_h tl_iface,
     return 1;
 }
 
+static ucs_status_t
+uct_tcp_iface_parse_virtual_dev(struct dirent *entry, void *ctx)
+{
+    static const char *low_level_dev_prefix = "lower_";
+    uct_tcp_iface_virtual_dev_path_t *data  = ctx;
+    size_t prefix_len;
+
+    prefix_len = ucs_string_common_prefix_len(entry->d_name,
+                                              low_level_dev_prefix);
+
+    if ((prefix_len == strlen(low_level_dev_prefix)) && !data->found) {
+        ucs_string_buffer_appendf(data->dev_path, "/%s", entry->d_name);
+        data->found = 1;
+    }
+
+    return UCS_OK;
+}
+
+static const char *
+uct_tcp_iface_get_sysfs_path(const char *dev_name, char *path_buffer)
+{
+    ucs_string_buffer_t dev_path          = UCS_STRING_BUFFER_INITIALIZER;
```

Comment:
can use UCS_STRING_BUFFER_ONSTACK with PATH_MAX length

Reply:
too big to fit in stack:
string_buff_elem * MAX_PATH > UCS_ALLOCA_MAX_SIZE (24 * 256 > 1200)